src/bpftrace.cpp

```diff
@@ -101,29 +102,15 @@ int BPFtrace::add_probe(ast::Probe &p)
 {
   for (auto attach_point : *p.attach_points)
   {
-    if (attach_point->provider == "BEGIN")
-    {
-      Probe probe;
-      probe.path = "/proc/self/exe";
-      probe.attach_point = "BEGIN_trigger";
-      probe.type = probetype(attach_point->provider);
-      probe.orig_name = p.name();
-      probe.name = p.name();
-      probe.loc = 0;
-      probe.index = attach_point->index(probe.name) > 0 ?
-          attach_point->index(probe.name) : p.index();
-      special_probes_.push_back(probe);
-      continue;
-    }
-    else if (attach_point->provider == "END")
-    {
+    if (attach_point->provider == "BEGIN" || attach_point->provider == "END") {
```

Comment:
Just one nitpick - could you leave the open bracket on a separate line as it was

Reply:
Yes, fixed.